segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/BookKeeperLog.java

```diff
@@ -396,13 +473,17 @@ private void handleWriteException(int responseCode, Write write) {
         try {
             if (ex instanceof BKException.BKLedgerFencedException) {
                 // We were fenced out.
-                ex = new DataLogWriterNotPrimaryException("Log not primary anymore.", ex);
+                ex = new DataLogWriterNotPrimaryException("BookKeeperLog is not primary anymore.", ex);
```

Comment:
One invariant that we need to preserve is that an old container owner never fences out a new container owner. When I read this change, it made me wonder whether there is any situation in which this invariant can be violated.